### PR TITLE
Use the eq-bytecode even without optimizations.

### PR DESCRIPTION
**Pull request by floitsch**
Also add more tests and a schedule that runs tests during the night.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -180,19 +190,25 @@ jobs:
...
-        if: runner.os == 'Linux' && github.event_name == 'push' && github.ref == 'refs/heads/master'
+      - name: Build test SDKS - Linux only
+        if: env.DO_MORE_TESTS && runner.os == 'Linux'
+        run: |
+          make HOST=host32 BUILD_TYPE=Debug sdk
+          make HOST=host-ctp TOOLCHAIN=host TOIT_CHECK_PROPAGATED_TYPES=1 sdk
+
+      - name: Build test SDKS
+        shell: bash
+        if: env.DO_MORE_TESTS
```
Comment (kasperl) [inline .github/workflows/ci.yml:205]:
```suggestion
        if: env.DO_MORE_TESTS == 'true'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -180,19 +190,25 @@ jobs:
...
-          fi
 
       - name: Ccache stats
         run: ccache -s
 
-      # Build 64-bit debug if we are comitting to master.
-      - name: Build 64-bit debug
-        if: runner.os == 'Linux' && github.event_name == 'push' && github.ref == 'refs/heads/master'
+      - name: Build test SDKS - Linux only
+        if: env.DO_MORE_TESTS && runner.os == 'Linux'
```
Comment (kasperl) [inline .github/workflows/ci.yml:198]:
```suggestion
        if: env.DO_MORE_TESTS == 'true' && runner.os == 'Linux'
```

File: src/compiler/CMakeLists.txt [hunk 3]
```diff
@@ -38,6 +38,17 @@ set_source_files_properties(propagation/type_propagator.cc PROPERTIES COMPILE_OP
 set_source_files_properties(propagation/type_set.cc PROPERTIES COMPILE_OPTIONS "-O3")
 set_source_files_properties(propagation/type_stack.cc PROPERTIES COMPILE_OPTIONS "-O3")
 
+if (DEFINED ENV{TOIT_OPTIMIZATION_OVERRIDE})
+  # Use the optimization level from the environment and pass it in as define.
+  message("OVERRIDING OPTIMIZATION LEVEL")
+  target_compile_definitions(toit_compiler PRIVATE TOIT_OPTIMIZATION_OVERRIDE=$ENV{TOIT_OPTIMIZATION_OVERRIDE})
+endif()
+if (DEFINED ENV{TOIT_ASSERT_OVERRIDE})
+  # Use the optimization level from the environment and pass it in as define.
```
Comment (kasperl) [inline src/compiler/CMakeLists.txt:47]:
```suggestion
  # Use the assert setting from the environment and pass it in as define.
```

File: src/compiler/compiler.cc [hunk 4]
```diff
@@ -1659,6 +1659,15 @@ Pipeline::Result Pipeline::run(List<const char*> source_paths, bool propagate) {
     Flags::enable_asserts = true;
   }
 
+#ifdef TOIT_OPTIMIZATION_OVERRIDE
+  configuration_.optimization_level = TOIT_OPTIMIZATION_OVERRIDE;
+#endif
+#if defined(TOIT_ASSERT_OVERRIDE) && TOIT_ASSERT_OVERRIDE == 0
+  Flags::enable_asserts = false;
+#elif TOIT_ASSERT_OVERRIDE == 1
```
Comment (kasperl) [inline src/compiler/compiler.cc:1667]: Not sure about this:

```suggestion
#elif TOIT_ASSERT_OVERRIDE != 0
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -180,19 +190,25 @@ jobs:
...
-            make HOST=host-ctp TOOLCHAIN=host TOIT_CHECK_PROPAGATED_TYPES=1 sdk
-          fi
 
       - name: Ccache stats
         run: ccache -s
 
-      # Build 64-bit debug if we are comitting to master.
-      - name: Build 64-bit debug
-        if: runner.os == 'Linux' && github.event_name == 'push' && github.ref == 'refs/heads/master'
+      - name: Build test SDKS - Linux only
```
[the remark this answers is not in the document]
Comment (floitsch) [reply .github/workflows/ci.yml]: done.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (floitsch) [reply .github/workflows/ci.yml]: done.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -180,19 +190,25 @@ jobs:
...
+        if: env.DO_MORE_TESTS && runner.os == 'Linux'
+        run: |
+          make HOST=host32 BUILD_TYPE=Debug sdk
+          make HOST=host-ctp TOOLCHAIN=host TOIT_CHECK_PROPAGATED_TYPES=1 sdk
+
+      - name: Build test SDKS
```
[the remark this answers is not in the document]
Comment (floitsch) [reply .github/workflows/ci.yml]: done.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (floitsch) [reply .github/workflows/ci.yml]: done.

File: src/compiler/CMakeLists.txt (hunk 3 above)
Comment (floitsch) [reply src/compiler/CMakeLists.txt]: done.

File: src/compiler/compiler.cc (hunk 4 above)
Comment (floitsch) [reply src/compiler/compiler.cc]: done.

**Pull request merged**